extension/content/components/pages/RecipeFormPage.js

```diff
@@ -41,7 +43,7 @@ export default function RecipeFormPage() {
       setData(INITIAL_RECIPE_DATA);
       setImportInstructions("");
     }
-  }, [recipeId, experimenterSlug]);
+  }, [recipeId, experimenterSlug, environmentKey]);
```

Comment:
As an aside, this effect function is getting awfully big, and might benefit from some refactoring.

Reply:
I'm not sure how I would meaningfully refactor this.

Reply:
I poked at it a bit and never mind, it's big but it's necessarily big. Carry on.